src/txmempool.cpp

```diff
@@ -790,8 +790,39 @@ bool CCoinsViewMemPool::HaveCoins(const uint256 &txid) const {
 
 size_t CTxMemPool::DynamicMemoryUsage() const {
     LOCK(cs);
-    // Estimate the overhead of mapTx to be 6 pointers + an allocation, as no exact formula for boost::multi_index_contained is implemented.
-    return memusage::MallocUsage(sizeof(CTxMemPoolEntry) + 6 * sizeof(void*)) * mapTx.size() + memusage::DynamicUsage(mapNextTx) + memusage::DynamicUsage(mapDeltas) + cachedInnerUsage;
+
+    size_t total = 0;
+
+    // Estimate the overhead of mapTx to be 6 pointers + an allocation, as no exact formula for
+    // boost::multi_index_contained is implemented.
+    total += memusage::MallocUsage(sizeof(CTxMemPoolEntry) + 6 * sizeof(void*)) * mapTx.size();
+
+    // Two metadata maps inherited from Bitcoin Core
+    total += memusage::DynamicUsage(mapNextTx) + memusage::DynamicUsage(mapDeltas);
+
+    // Saves iterating over the full map
+    total += cachedInnerUsage;
+
+    // Wallet notification
+    total += memusage::DynamicUsage(mapRecentlyAddedTx);
+
+    // Nullifier set tracking
+    total += memusage::DynamicUsage(mapSproutNullifiers) + memusage::DynamicUsage(mapSaplingNullifiers);
+
+    // DoS mitigation
+    total += memusage::DynamicUsage(recentlyEvicted) + memusage::DynamicUsage(weightedTxTree);
+
+    // Insight-related structures
+    size_t insight;
+    insight += memusage::MallocUsage(sizeof(CMempoolAddressDeltaKey) +
+        sizeof(CMempoolAddressDelta)) + memusage::DynamicUsage(mapAddress);
+    insight += memusage::DynamicUsage(mapAddressInserted);
+    insight += memusage::MallocUsage(sizeof(CSpentIndexKey) +
+        sizeof(CSpentIndexValue)) + memusage::DynamicUsage(mapSpent);
+    insight += memusage::DynamicUsage(mapSpentInserted);
+    total =+ insight;
```

Comment:
This is the change I was meaning:
```suggestion
    // Insight-related structures
    size_t insight;
    insight += memusage::DynamicUsage(mapAddress);
    insight += memusage::DynamicUsage(mapAddressInserted);
    insight += memusage::DynamicUsage(mapSpent);
    insight += memusage::DynamicUsage(mapSpentInserted);
    total =+ insight;
```

Reply:
Nice, thanks :)